### PR TITLE
fix: update TypeScript declaration for KV

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

update TypeScript declaration for KV

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: types/kv-model.d.ts [hunk 1]
```diff
@@ -164,5 +194,8 @@ export declare class KeyValueModel extends ModelBase {
    *
    * @header KeyValueModel.iterateKeys(filter)
    */
-  iterateKeys(filter?: Filter, options?: Options): Iterator<Promise<string>>;
+  static iterateKeys(
+    filter?: Filter,
+    options?: Options,
+  ): Iterator<Promise<string>>;
```
Comment (raymondfeng) [inline types/kv-model.d.ts]: This is not accurate. It should be `AsyncIterator<string>` which is only available with `esnext`. See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html.

```
Caveats
Keep in mind that our support for async iterators relies on support for Symbol.asyncIterator to exist at runtime. You may need to polyfill Symbol.asyncIterator, which for simple purposes can be as simple as: (Symbol as any).asyncIterator = Symbol.asyncIterator || Symbol.for("Symbol.asyncIterator");
You also need to include esnext in your --lib option, to get the AsyncIterator declaration if you do not already have it.
Finally, if your target is ES5 or ES3, you’ll also need to set the --downlevelIterators flag.
```

File: types/kv-model.d.ts [hunk 2]
```diff
@@ -49,13 +49,43 @@ export declare class KeyValueModel extends ModelBase {
...
+  /**
+   * Delete all keys (and values) associated to the current model.
+   *
+   * @options {Object} options Unused ATM, placeholder for future options.
+   * @callback {Function} callback
+   * @param {Error} err Error object.
+   * @promise
+   */
+  static deleteAll(
+    key: string,
```
Comment (virkt25) [inline types/kv-model.d.ts]: Don't think it's supposed to take a `key` argument for `deleteAll`.

Comment (raymondfeng) [reply types/kv-model.d.ts]: Good catch. Will fix.

File: types/kv-model.d.ts [hunk 3]
```diff
@@ -114,7 +144,7 @@ export declare class KeyValueModel extends ModelBase {
    *
    * @header KeyValueModel.keys(filter, cb)
    */
-  keys(
+  static keys(
     filter?: Filter,
```
Comment (bajtos) [inline types/kv-model.d.ts]: `KVFilter`, not `Filter`.

**Pull request closed without merging**